src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ObjectModelConverters.cs

```diff
@@ -34,7 +49,13 @@ public static TestNode ToTestNode(this TestCase testCase, bool isTrxEnabled, ICl
             DisplayName = displayNameFromTestResult ?? testCase.DisplayName ?? testCase.FullyQualifiedName,
         };
 
-        CopyVSTestProperties(testCase.Properties, testNode, testCase, testCase.GetPropertyValue, isTrxEnabled, client);
+        // This will be false for Expecto and NUnit currently, as they don't provide ManagedType/ManagedMethod.
+        if (TryGetMethodIdentifierProperty(testCase, out TestMethodIdentifierProperty? methodIdentifierProperty))
+        {
+            testNode.Properties.Add(methodIdentifierProperty);
+        }
+
+        CopyVSTestProperties(testCase.Properties, testNode, testCase, testCase.GetPropertyValue, isTrxEnabled, ShouldAddVSTestProviderProperties(serviceProvider));
```

Comment:
If nunit provides both managed and vstest properties (we cannot stop them), it will combine both. 

We can have something like this to ensure only one type of properties are populated,
```
if (hasManagedProperties)
   addManaged
else
   addVstest
```

Reply:
`MSTestDiscoverer.TestCategory` needs to be copied regardless. It could be separated though.

Reply:
A suggestion then. Maybe we can rename CopyVSTestProperties to something generic CopyProperties and then pass different flags `ShouldAddVSTestProviderProperties`, `ShouldCopySomething`.

Reply:
The properties are still "VSTest" properties though. I went ahead and extracted the VSTest properties that we always copy (i.e, those not specific to vstestProvider capabilitiy) outside of this method, renamed the method to CopyVSTestProviderProperties, and removed the `addVSTestProviderProperties` parameter and made it a caller check. I don't know if it makes it harder to review though.